src/AiClient.php

```diff
@@ -0,0 +1,413 @@
+    /**
+     * Creates a new message builder for fluent API usage.
+     *
+     * This method will be implemented once MessageBuilder is available.
+     * MessageBuilder will provide a fluent interface for constructing complex
+     * messages with multiple parts, attachments, and metadata.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|null $text Optional initial message text.
+     * @return object MessageBuilder instance (type will be updated when MessageBuilder is available).
+     *
+     * @throws \RuntimeException When MessageBuilder is not yet available.
+     */
+    public static function message(?string $text = null)
+    {
+        throw new \RuntimeException(
+            'MessageBuilder is not yet available. This method depends on builder infrastructure. ' .
+            'Use direct generation methods (generateTextResult, generateImageResult, etc.) for now.'
+        );
+    }
+
+    /**
+     * Template method for executing generation operations.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @param string $type The generation type.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    private static function executeGeneration($prompt, ?ModelInterface $model, string $type): GenerativeAiResult
+    {
+        // Convert prompt to standardized Message array format
+        $messages = PromptNormalizer::normalize($prompt);
+        /** @var list<Message> $messageList */
+        $messageList = array_values($messages);
+
+        // Map type to specific methods
+        switch ($type) {
+            case 'text':
+                $resolvedModel = $model ?? ModelDiscovery::findTextModel(self::defaultRegistry());
+                InterfaceValidator::validateTextGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateTextResult($messageList);
+
+            case 'image':
+                $resolvedModel = $model ?? ModelDiscovery::findImageModel(self::defaultRegistry());
+                InterfaceValidator::validateImageGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateImageResult($messageList);
+
+            case 'speech':
+                $resolvedModel = $model ?? ModelDiscovery::findSpeechModel(self::defaultRegistry());
+                InterfaceValidator::validateSpeechGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateSpeechResult($messageList);
+
+            default:
+                throw new \InvalidArgumentException("Unsupported generation type: {$type}");
+        }
```

Comment:
See my comment above: This method does something that `PromptBuilder` terminate methods should be doing. I'm wary we're duplicating too many efforts here. `AiClient` should merely be the API entrypoint, and an easy to use wrapper around the heavy logic happening in `PromptBuilder`.

Reply:
I'm kind of tempted to leave some of the logic in this PR, such as discovering the model to use, and then refactoring it in the subsequent Prompt Builder PR to have the `generateX` functions use the builder.

Reply:
OK, that works for me - I wasn't clear about this sequence. I was assuming the `PromptBuilder` PR would come first, as `AiClient` is using it, rather than the other way around. But either way works, as long as we're all on the same page.